strongbox-resources/strongbox-common-resources/src/main/resources/logback.xml

```diff
@@ -8,28 +8,51 @@
     </contextListener>
 
     <appender name="CONSOLE" class="ch.qos.logback.core.ConsoleAppender">
+
+        <filter class="ch.qos.logback.core.filter.EvaluatorFilter">
+            <evaluator>
+                <expression>return mdc.containsKey("strongbox-cron-context-name");</expression>
+            </evaluator>
+            <OnMismatch>NEUTRAL</OnMismatch>
+            <OnMatch>DENY</OnMatch>
+        </filter>
+
         <encoder>
             <pattern>%d{HH:mm:ss.SSS dd-MM-yyyy} | %-5.5p | %-20.20t | %-50.50logger{50} | %m%n</pattern>
         </encoder>
     </appender>
 
-    <appender name="FILE" class="ch.qos.logback.core.rolling.RollingFileAppender">
-        <file>${strongbox.vault}/logs/strongbox.log</file>
-        <rollingPolicy class="ch.qos.logback.core.rolling.SizeAndTimeBasedRollingPolicy">
-            <fileNamePattern>strongbox_%d{yyyy-MM-dd}.%i.log</fileNamePattern>
+    <appender name="FILE" class="ch.qos.logback.classic.sift.SiftingAppender">
 
-            <fileNamePattern>strongbox-%d{yyyy-MM-dd}.%i.log</fileNamePattern>
-            <maxFileSize>128MB</maxFileSize>
-            <maxHistory>31</maxHistory>
-            <totalSizeCap>1GB</totalSizeCap>
+        <discriminator
+            class="ch.qos.logback.classic.sift.MDCBasedDiscriminator">
```

Comment:
Would you mind enlightening me what this does?

Reply:
this produce log files based on Discriminator value, which is stored in ThreadLocal

Reply:
this explained [here](https://logback.qos.ch/manual/appenders.html#SiftingAppender)